lib/analyse.ml

```diff
@@ -191,7 +195,14 @@ module Analysis = struct
             Current.Process.check_output ~cwd:dir ~cancellable:true ~job cmd >>= begin function
               | Error _ ->
                 (* new release *)
-                Lwt_result.return (add_pkg ~path ~name ~package New pkgs)
+                let package_dir = "packages/"^name in
+                let cmd = "", [| "git"; "show"; master^":"^package_dir |] in
+                Current.Process.check_output ~cwd:dir ~cancellable:true ~job cmd >>= begin function
+                | Ok _ -> Lwt_result.return (add_pkg ~path ~name ~package NewRelease pkgs)
+                | Error _ ->
+                  (* newly published package *)
+                  Lwt_result.return (add_pkg ~path ~name ~package NewPackage pkgs)
```

Comment:
I suggest retaining the named helper function from the original code in lint, as it makes the function and intent of the logic here much easier to scan IMO. Since we are only checking for `Ok _` or `Error _` without any consideration of the content, this is isomorphic to a boolean check, and so making that explicit communicates the intent more concisely. Moreover, a named function here also makes the comment on 203 unnecessary.

Reply:
This makes, sense. Thanks!